### ISSUE TITLE
Swipe to archive inconsistent with long-press + archive

### ISSUE THREAD

**Issue by floatingatoms** [signalapp/Signal-Android#13631]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Swiping to archive marks an unread message as read. This is inconsistent with the behaviour when using long press + archive, which leaves an unread message as unread. Whatever the method of archiving, it should have consistent behaviour regarding the read status and ideally should leave an unread message as unread across all methods.

### Steps to reproduce
- have an unread message in the message list and do not open it
- swipe to archive
- go to archived chats list
- message has been put to "read"

**Actual result:** the unread and unopened message is marked as read after swiping to archive
**Expected result:** read status should be unaffected and leave unread messages as unread after swiping to achive, just like when doing long press + archive

### Device info

**Device:** Pixel 7a
**Android version:** 14
**Signal version:** 7.11.4

**Comment by trysten** [comment]
tl;dr: Messages shouldn't be marked as read until the user does some action adjacent to reading them. Mark messages as read when they are opened, not when they are archived.

The menu action has the correct behavior: archiving the message without marking the messages as read or consequently sending read receipts. The swipe behavior is incorrect: marking messages the user has not opened or read as read and sending read receipts when the message has not been read. This isn't just an inconsistency, it's user-hostile behavior. There has already been at least [one horror story](https://github.com/signalapp/Signal-Android/issues/9147#issuecomment-803044428) of a user being stalked because of this bug. 

Why is this being [ignored](https://github.com/signalapp/Signal-Android/issues/9147#issuecomment-1802435881)?

**Comment by floatingatoms** [comment]
I agree that the menu action has the correct behaviour, archiving something shouldn't affect read/unread status of a message in my view, don't think anyone would expect that to happen.

In the other issue you link, I find it surprising to read that @greyson-signal got it wrong in his [comment](https://github.com/signalapp/Signal-Android/issues/9147#issuecomment-555321568) because actually, it is inconsistent *within* the Android app itself.. would be good to get some response as surely this inconsistency is a bug.

I noticed someone else on the iOS issue saying they had stopped their donations because of this and I must admit this is a reason why I have not restarted my donations (I stopped for other reasons). I was on the 10 GBP a month tier too for a while. It feels really weird for Signal to make such a point of this incredibly reasonable request that literally has no negative consequence at all.. If they really want donations to support them long term, surely engaging positively with your donors/customer should be a priority?

**Comment by greyson-signal** [comment]
This has been addressed in 7.24.2
